### PR TITLE
docs: add documentation for Google Cloud database integrations

### PR DESCRIPTION
**Pull request by averikitsch**
**Description:** add documentation for Google Cloud database integrations
**Issue:** NA
**Dependencies:** NA

### CODE REVIEW

**Review by efriis — COMMENTED** [review]
Can we

- prefix all the page titles (the initial `# Title` line in each ipynb/mdx file) with "Google" so they show up next to each other in the lists?
- use `-` instead of `_` in the `pip install` commands (e.g. `pip install langchain-google-cloud-sql-pg`)
- link some of the boilerplate in the provider page (right now each page has "Set your Google Cloud Project") - would it be better to just write these instructions once, and link them from each page?

Happy to discuss any of the above as well!

File: docs/docs/integrations/document_loaders/alloydb.ipynb [hunk 1]
```diff
@@ -0,0 +1,378 @@
+{
+ "cells": [
+  {
+   "cell_type": "markdown",
+   "metadata": {
+    "id": "E_RJy7C1bpCT"
+   },
+   "source": [
+    "# Google AlloyDB for PostgreSQL\n",
```
Comment (leo-gan) [inline docs/docs/integrations/document_loaders/alloydb.ipynb]: Please name files with a `file name` equal to the `Title` of the notebook.
In this way, the Navbar (ToC on the left side of the screen) is properly sorted. It is sorted by the file names!
The title is the H1 header ("# Google AlloyDB for PostgreSQL" in your case).
As I can see this should be applied to all other notebooks.
Thanks!

Comment (averikitsch) [reply docs/docs/integrations/document_loaders/alloydb.ipynb]: done!

**Comment by averikitsch** [comment]
> Can we
> 
> * prefix all the page titles (the initial `# Title` line in each ipynb/mdx file) with "Google" so they show up next to each other in the lists?
> * use `-` instead of `_` in the `pip install` commands (e.g. `pip install langchain-google-cloud-sql-pg`)
> * link some of the boilerplate in the provider page (right now each page has "Set your Google Cloud Project") - would it be better to just write these instructions once, and link them from each page?
> 
> Happy to discuss any of the above as well!

First two items completed. I would love to follow up with refactoring of the boilerplate.

**Comment by leo-gan** [comment]
Awesome! @averikitsch  Thanks a lot for these integrations!

**Comment by averikitsch** [comment]
Ready to merge :) Thank you!

File: docs/docs/integrations/platforms/google.mdx [hunk 2]
```diff
@@ -121,6 +121,67 @@ from langchain_google_vertexai import VertexAIModelGarden
...
+
+> [AlloyDB](https://cloud.google.com/alloydb) is a fully managed relational database service that offers high performance, seamless integration, and impressive scalability on Google Cloud. AlloyDB is 100% compatible with PostgreSQL.
+
+Install the python package:
+
+```bash
+pip install langchain-google-alloydb-pg
+```
+
+See [usage example](/docs/integrations/vectorstores/alloydb).
```
Comment (leo-gan) [inline docs/docs/integrations/platforms/google.mdx:134]: Links are equal to the file names. It should be
`(/docs/integrations/vectorstores/google_alloydb)`
The same for other links of the usage examples below.

**Review by leo-gan — CHANGES_REQUESTED** [review]
nits in the platforms/google.mdx

File: docs/docs/integrations/platforms/google.mdx [hunk 3]
```diff
@@ -521,20 +719,142 @@ from langchain_community.agent_toolkits import GmailToolkit
 
 ## Memory
 
-### Firestore
+### AlloyDB for PostgreSQL
```
[the remark this answers is not in the document]
Comment (averikitsch) [reply docs/docs/integrations/platforms/google.mdx:722]: Within the Google provider page, I don't think we need to prefix with Google.

**Review by leo-gan — APPROVED** [review]
lgtm
Thanks a lot!

File: docs/docs/integrations/platforms/google.mdx (hunk 2 above)
Comment (averikitsch) [reply docs/docs/integrations/platforms/google.mdx]: fixed! Thanks for catching

**Review by efriis — APPROVED** [review]

**Pull request merged**